server/src/routes/signin.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * A router that handles user signin process
+ * It responds to POST requests on endpoint of "/api/auth/signin"
+ * This router takes in user "email" and "password" as its arguments
+ * and generates a JWT for the user object that contains "id" and "email" field inside the payload
+ * The jwt is then sent to user in form of cookies
+ */
+
+const router = require("express").Router();
+const { body } = require("express-validator");
+const db = require("../../database/models/index");
+const Password = require("../utils/password");
+const BadRequestError = require("../errors/bad-request-error");
+const validateRequest = require("../middleware/validate-request");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+router.post(
+  "/api/auth/signin",
```

Comment:
I think for our routes, we don't need to use /api/auth as a prefix, maybe just /api/